apis/core/v1beta1/flagsourceconfiguration_types.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	SyncProviderKubernetes SyncProviderType = "kubernetes"
+	SyncProviderFilepath   SyncProviderType = "filepath"
+	SyncProviderHttp       SyncProviderType = "http"
+	SyncProviderGrpc       SyncProviderType = "grpc"
+	SyncProviderFlagdProxy SyncProviderType = "flagd-proxy"
+)
+
+type SyncProviderType string
+
+// FlagSourceConfigurationSpec defines the desired state of FlagSourceConfiguration
+type FlagSourceConfigurationSpec struct {
+	// MetricsPort defines the port to serve metrics on, defaults to 8014
+	// +optional
+	MetricsPort int32 `json:"metricsPort"`
+
+	// Port defines the port to listen on, defaults to 8013
+	// +optional
+	Port int32 `json:"port"`
+
+	// SocketPath defines the unix socket path to listen on
+	// +optional
+	SocketPath string `json:"socketPath"`
+
+	// Evaluator sets an evaluator, defaults to 'json'
+	// +optional
+	Evaluator string `json:"evaluator"`
+
+	// Image allows for the sidecar image to be overridden, defaults to 'ghcr.io/open-feature/flagd'
+	// +optional
+	Image string `json:"image"`
+
+	// Tag to be appended to the sidecar image, defaults to 'main'
+	// +optional
+	Tag string `json:"tag"`
```

Comment:
This is related to issue https://github.com/open-feature/open-feature-operator/issues/517 

This can be considered as a security concern and hence we can discuss on removing these 

cc - @toddbaert  @thisthat appreciate your opinion

Reply:
adapted the code to remove tag and image from `FlagSourceConfiguration` CRD

Reply:
Ya I'm fine with removing them. I have only heard agreements on this.

Reply:
With further implementation I came across some issues regarding this. Read comment here https://github.com/open-feature/open-feature-operator/issues/517#issuecomment-1770941958

I would suggest that covering this should be part of a separate PR, as it will bump the complexity of the PR, which will be adapting the code to use `v1beta1` instead of `v1alpha1`. Any objections on this ?

Reply:
I agree we can adapt to use `v1beta1` in a new PR, once we have that. OFO is locked to a particular version of flagd, so we shouldn't have breakage there.

It will be a breaking change in flagd though, so we should note that (with a `feat!`) so things are versioned accordingly.